pod.go

```diff
@@ -525,12 +528,108 @@ func (p *Pod) createSetStates() error {
 	return nil
 }
 
+type assetType string
+
+const (
+	kernelAsset assetType = "kernel"
+	imageAsset            = "image"
+)
+
+func validateAsset(path string, t assetType) bool {
+	if !filepath.IsAbs(path) {
+		return false
+	}
+
+	// TODO Check for asset type
+	switch t {
+	case kernelAsset:
+		return true
+	case imageAsset:
+		return true
+	}
+
+	return true
+}
+
+func selectAssetFromAnnotation(podConfig *PodConfig, pathAnnotation, hashAnnotation string) (string, error) {
+	assetPath, ok := podConfig.Annotations[pathAnnotation]
+	if !ok {
+		return "", nil
+	}
+
+	if !filepath.IsAbs(assetPath) {
+		return "", fmt.Errorf("%s is not an absolute path", assetPath)
+	}
+
+	assetHashString, ok := podConfig.Annotations[hashAnnotation]
+	if !ok {
+		return assetPath, nil
+	}
+
+	// We have a hash annotation, we need to verify it
+	assetBytes, err := ioutil.ReadFile(assetPath)
+	if err != nil {
+		return "", err
+	}
+
+	if len(assetHashString) != sha512.Size {
+		return "", fmt.Errorf("Invalid hash for %s", assetPath)
+	}
+
+	assetHashComputed := sha512.Sum512(assetBytes)
+
+	var assetHash [sha512.Size]byte
+	copy(assetHash[:], assetHashString)
+
+	if assetHashComputed != assetHash {
+		return "", fmt.Errorf("Invalid hash for %s", assetPath)
+	}
+
+	return assetPath, nil
+}
+
+func selectAssets(podConfig *PodConfig) error {
+	kernelPath, err := selectAssetFromAnnotation(podConfig, annotations.KernelPath, annotations.KernelHash)
+	if err != nil {
+		return err
+	}
+
+	imagePath, err := selectAssetFromAnnotation(podConfig, annotations.ImagePath, annotations.ImageHash)
+	if err != nil {
+		return err
+	}
+
+	// Do we want to update the hypervisor config only if both
+	// paths are not empty?
+	if kernelPath != "" {
+		podConfig.HypervisorConfig.KernelPath = kernelPath
+	}
+
+	if imagePath != "" {
+		podConfig.HypervisorConfig.ImagePath = imagePath
+	}
+
+	if !validateAsset(kernelPath, kernelAsset) {
```

Comment:
Wouldn't it be better if these `validateAsset()` calls came before the change to `podConfig`?